src/operators/permute.c

```diff
@@ -27,6 +27,38 @@
 
 #include "graph/attributes.h"
 
+/**
+ * \brief Inverts a permutation.
+ *
+ * Produces the inverse of \p permutation into \p inverse and at the same time it checks
+ * that the permutation vector is valid, i.e. all indices are within range and there are
+ * no duplicate entries.
+ *
+ * \param permutation A permutation vector containing 0-based integer indices.
+ * \param inverse An initialized vector. The inverse of \p permutation will be stored here.
+ * \return Error code.
+ */
+static int igraph_i_invert_permutation(const igraph_vector_t *permutation, igraph_vector_t *inverse) {
```

Comment:
I wonder whether we should expose this function in public. It seems useful enough for end-users who might have an R1-style permutation for an R2-style igraph function or vice versa.

Reply:
Yes, that's the intention, but not in `master`.